### PR TITLE
Database connection, request validation

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: api/schema.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	schema := schemaObj[reqRes][loc]
+	if schema != nil {
+		schema["type"] = "object"
+		schema["additionalProperties"] = false
+		keys := getKeys(schema["properties"])
+		if keys != nil {
+			schema["required"] = keys
+		}
+	}
+}
```
Comment (spaced4ndy) [inline api/schema.go:38]: that's probably a stupid question, but what does `jtdToJsonSchema` do? i.e. it doesn't return anything, is `schemaObj` modified by reference somehow (which is also not clear for me)?

File: api/handler.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+		schema.Validate("", body, &errors)
+
+		if len(errors) > 0 {
+			log.Println("Invalid data:", errors)
+			w.WriteHeader(http.StatusBadRequest)
+			io.WriteString(w, "Bad Request")
+			return
+		}
+
+		log.Println(body["recipient"])
```
Comment (spaced4ndy) [inline api/handler.go]: is this for debugging only?

Comment (epoberezkin) [reply api/handler.go]: yes

File: api/schema.go (hunk 1 above)
Comment (epoberezkin) [reply api/schema.go:38]: jtdToJsonSchema converts from JTD/JDDF (new spec, JSON Type Definition) to JSON Schema  that is more verbose and there is a library for it. I couldn't make go library for JDDF work.

Comment (epoberezkin) [reply api/schema.go:38]: yes, maps and slices are passed by reference in go.

**Review by spaced4ndy — APPROVED** [review]

**Pull request merged**